evolver/base.py

```diff
@@ -215,35 +265,45 @@ class BaseInterface(ABC):
     class Config(BaseConfig): ...
 
     @classmethod
-    def create(cls, config: ConfigDescriptor | dict | str | None = None):
+    def create(cls, config: ConfigDescriptor | BaseConfig | dict | str | None = None):
         """Create an instance from a config."""
 
-        def validate_descriptor(descriptor: ConfigDescriptor):
-            if not issubclass(cls, descriptor.classinfo):
+        def validate_classinfo(classinfo: type | str):
+            classinfo = import_string(classinfo) if isinstance(classinfo, str) else classinfo
+            if not issubclass(classinfo, cls):  # TODO: don't allow polymorph?
                 raise TypeError(
-                    f"The given {ConfigDescriptor.__name__} for '{descriptor.classinfo}' is not compatible "
+                    f"The given {ConfigDescriptor.__name__} for '{classinfo}' is not compatible "
                     f"with this class '{cls.__qualname__}'"
                 )
-            return cls.Config.model_validate(descriptor, context=dict(extra="forbid"))
 
-        # We first try to create a valid Config instance and then from that create an instance of cls.
+        # We first create a validated Config instance and then use that to create an instance of cls.
         if config is None:
             # Empty config.
             config = cls.Config()
+        elif isinstance(config, _BaseConfig):
+            pass  # Already a config instances so do nothing.
         elif isinstance(config, ConfigDescriptor):
-            config = validate_descriptor(config)
+            validate_classinfo(config.classinfo)
+            config = config.config  # This has already been validated but is still just a dict.
+        elif _is_descriptor_dict(config):
+            validate_classinfo(config["classinfo"])
+            config = cls.Config.model_validate(config["config"])
+        elif isinstance(config, dict):
+            config = cls.Config.model_validate(config)
         else:
-            # Next handle dict | str, which could be that representing a Config or a ConfigDescriptor.
-            # First see if config is actually a descriptor by trying to create one, since it's only fields are limited
-            # to classinfo and config.
+            # Last is a str representing either a Config or ConfigDescriptor.
             try:
                 descriptor = ConfigDescriptor.model_validate(config, context=dict(extra="forbid"))
-                config = validate_descriptor(descriptor)
             except pydantic.ValidationError:
                 config = cls.Config.model_validate(config)
+            else:
+                validate_classinfo(descriptor.classinfo)
+                config = descriptor.config
+
+        config_dict = config.shallow_model_dump() if isinstance(config, _BaseConfig) else config
```

Comment:
It might be worth a note here for posterity as to why this is needed - is it because this could be a dict due to 287 or 301? In the latter case, should it be validated or has it already been?

Reply:
Comment added. Also, note that this should be checked against ``_BaseModel`` and not ``_BaseConfig``.

Reply:
...and it will have been validated by then.